client/express-checkout/blocks/hooks/use-express-checkout.js

```diff
@@ -58,22 +59,46 @@ export const useExpressCheckout = ( {
 				return;
 			}
 
+			const shippingAddressRequired = shippingData?.needsShipping;
+
+			let shippingRates;
+			if ( shippingAddressRequired ) {
+				const hasValidRates =
+					shippingData?.shippingRates[ 0 ]?.shipping_rates?.length >
+					0;
+
+				if ( hasValidRates ) {
+					shippingRates = shippingData.shippingRates[ 0 ].shipping_rates.map(
+						( rate ) => {
+							return {
+								id: rate.rate_id,
+								amount: parseInt( rate.price, 10 ),
+								displayName: rate.name,
+							};
+						}
+					);
+				} else {
+					shippingRates = [
+						{
+							id: 'pending',
+							displayName: __(
+								'Pending',
+								'woocommerce-payments'
+							),
+							amount: 0,
+						},
+					];
+				}
+			}
```

Comment:
I'm just thinking aloud here, so I appreciate you indulging me. If `shippingAddressRequired` is not true, we will pass an empty `shippingRates` variable into the ECE options. But AFAICT, [Stripe's documentation](https://docs.stripe.com/js/elements_object/express_checkout_element_click_event#express_checkout_element_on_click-handler-resolve-shippingAddressRequired) suggests that `shippingRates` is required only when `shippingAddressRequired` is `true`. 

Presumably if `shippingAddressRequired` is not required, it isn't optional and it's irrelevant to offer any rates at all...which makes sense! Just double-checking, in case I have predictably misunderstood this time.